Mathlib/Topology/Algebra/Valued/LinearTopology.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Algebra.Valued.ValuedField
+
+/-!
+# Valuation rings of valued fields have a linear topology
+
+## Main Results
+* `IsLinearTopology.of_valued`: for a valued field `K`,
+  the valuation ring `𝒪[K]` has a linear topology
+
+-/
+open Valued Filter Topology
+
+variable {K Γ₀ : Type*} [Field K] [LinearOrderedCommGroupWithZero Γ₀] [Valued K Γ₀]
+
+instance IsLinearTopology.of_valued :
+    IsLinearTopology 𝒪[K] 𝒪[K] := by
+  by_cases hd : DiscreteTopology K
+  · have : DiscreteTopology 𝒪[K] := inferInstance
+    infer_instance
+  rw [isLinearTopology_iff_hasBasis_ideal]
+  have := (hasBasis_nhds_zero K Γ₀).comap (Subtype.val : 𝒪[K] → K)
+  have hn0 : 𝓝 (0 : 𝒪[K]) = comap Subtype.val (𝓝 0) := nhds_induced Subtype.val 0
+  rw [← hn0] at this
+  refine this.to_hasBasis ?_ ?_
+  · intro y _
+    let I : Set 𝒪[K] := {x : 𝒪[K] | Valued.v (x : K) < y}
+    lift I to Ideal 𝒪[K] with I' hI'
+    · simp only [Set.mem_setOf_eq, ZeroMemClass.coe_zero, map_zero, Subring.coe_add,
```

Comment:
This seems to be showing that the subset of `𝒪[K] ` of valuation `< y` is an ideal, for any `y`, and that sounds like something worth splitting off into a separate lemma (& likewise the statement that the corresponding subset of `K` is an `𝒪[K]`-submodule).